lib/vrl/stdlib/src/split.rs

```diff
@@ -3,7 +3,11 @@ use vrl::prelude::*;
 
 fn split(value: Value, limit: Value, pattern: Value) -> Resolved {
     let string = value.try_bytes_utf8_lossy()?;
-    let limit = limit.try_integer()? as usize;
+    let limit = match limit.try_integer()? {
+        x if x < 0 => return Err(format!("limit requires a positive integer got {}", x).into()),
```

Comment:
Alternatively we could [treat negative numbers as 0 like `truncate` does](https://github.com/vectordotdev/vector/blob/master/lib/vrl/stdlib/src/truncate.rs#L7) if that is preferred